src/pages/Admin/VenueOwnersModal.tsx

```diff
@@ -1,66 +1,101 @@
-import React, { useMemo, useState, useCallback } from "react";
-import { Modal } from "react-bootstrap";
-import { useFirestoreConnect } from "react-redux-firebase";
-import { useSelector } from "hooks/useSelector";
-import Fuse from "fuse.js";
+import React, { useMemo, useState, useCallback, useEffect } from "react";
+import { useFirestore } from "react-redux-firebase";
+import { Modal, FormControl } from "react-bootstrap";
 import { debounce } from "lodash";
-import { FormControl } from "react-bootstrap";
-import { User } from "types/User";
-import "./VenueOwnerModal.scss";
+
+import { DEFAULT_PARTY_NAME, DEFAULT_PROFILE_IMAGE } from "settings";
 import { AnyVenue } from "types/Firestore";
+import { User } from "types/User";
 import { WithId } from "utils/id";
-import { DEFAULT_PARTY_NAME, DEFAULT_PROFILE_IMAGE } from "settings";
 import { addVenueOwner, removeVenueOwner } from "api/admin";
 
+import "./VenueOwnerModal.scss";
+
+interface PartitionedOwnersOthers {
+  owners: WithId<User>[];
+  others: WithId<User>[];
+}
+const makePartitionOwnersFromOthersReducer = (ownerIds: string[]) => (
+  { owners, others }: PartitionedOwnersOthers,
+  user: WithId<User>
+) => {
+  if (ownerIds.includes(user.id)) {
+    return { owners: [...owners, user], others };
+  } else {
+    return { owners, others: [...others, user] };
+  }
+};
+
+const emptyPartition: PartitionedOwnersOthers = {
+  owners: [],
+  others: [],
+};
+
+const makePartyNameFilter = (searchText: string) => (user: WithId<User>) =>
+  user.partyName?.toLowerCase()?.includes(searchText.toLowerCase());
+
 interface VenueOwnersModalProps {
   visible: boolean;
   venue: WithId<AnyVenue>;
   onHide?: () => void;
 }
 
-export const VenueOwnersModal: React.FC<VenueOwnersModalProps> = (props) => {
-  const { visible, onHide, venue } = props;
+export const VenueOwnersModal: React.FC<VenueOwnersModalProps> = ({
+  visible,
+  onHide,
+  venue,
+}) => {
+  // Fetch all users the first time this component loads
+  // @debt reading every user is obviously bad.
+  const firestore = useFirestore();
+  useEffect(() => {
+    firestore
+      .collection("users")
+      .get()
+      .then((result) =>
+        result.docs.map<WithId<User>>(
+          (doc) => ({ ...doc.data(), id: doc.id } as WithId<User>) // TODO: be less hacky with types here?
+        )
```

Comment:
Calling this out as it feels kind of hacky, but wasn't sure if there was a better way to go about it.

A better way would probably be to build a custom validation function with type guards/similar:

- https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
It at least loads only once. This is the sort of place I have been tempted to go to a Firebase Function - but you can see the disastrous outcome of that with `getOnlineStats` where it takes forever to load and the CPU cycles cost us $. It seems like Firestore should offer a graphql api for use cases like this, especially as we move into 10,000s of users. It's at least not worse than the previous implementation, which also receives snapshots every time a new account is created.

Reply:
This specific comment was calling out my explicit typecasting with `as WithId<User>`

RE: Firebase functions, _nods_ makes sense. We can look further at tuning costs/etc when we have some spare dev time (lol@spare time ;p)